projects/plugins/crm/includes/ZeroBSCRM.AdminPages.Checks.php

```diff
@@ -170,7 +170,7 @@ function zeroBSCRM_isAPIRequest() {
 	// below is more reliable as QUERY_STRING will always be set for API requests.
 
 	// lazy, non-wp way of doing this
-	if ( isset( $_SERVER['QUERY_STRING'] ) && strpos( '#' . $_SERVER['QUERY_STRING'], 'api_key=zbscrm_' ) > 0 ) {
+	if ( isset( $_SERVER['QUERY_STRING'] ) && ( strpos( '#' . sanitize_text_field( wp_unslash( $_SERVER['QUERY_STRING'] ) ), 'api_key=zbscrm_' ) > 0 || strpos( '#' . sanitize_text_field( wp_unslash( $_SERVER['QUERY_STRING'] ) ), 'api_key=jpcrm_' ) ) ) {
```

Comment:
**Legacy code - for reference:** Wondering why we are not using $_GET to check `api_key` and then if it starts with `jpcrm_` or `zbscrm_`.

Reply:
Which is somewhat how we handle it in `jpcrm_is_api_request_authorised()`...

Based on the comments, it seems maybe it was trying to address some rewrite/permalinks oddities. But ultimately, a legacy oddity.

Reply:
This function only checks whether the input resembles an API request and is more lightweight compared to jpcrm_is_api_request_authorised.

Reply:
Indeed, I'm primarily referring to the overly-complex checks in `zeroBSCRM_isAPIRequest()` vs. `jpcrm_is_api_request_authorised()`. But the reason is simply "because it's legacy".